dotCMS/src/integration-test/java/com/dotmarketing/common/reindex/ReindexQueueAPITest.java

```diff
@@ -0,0 +1,121 @@
+
+        //cleans up data first
+        reindexQueueAPI.deleteReindexAndFailedRecords();
+
+        try {
+            //adds a new entry and verifies it is not marked as failed
+            reindexQueueAPI.addReindexHighPriority(recordId);
+            List<ReindexEntry> failedRecords = reindexQueueAPI.getFailedReindexRecords();
+            assertFalse(UtilMethods.isSet(failedRecords));
+
+            //gets ReindexEntry from recordId
+            final ReindexEntry currentEntry = getReindexEntry(recordId);
+
+            //marks the entry as failed and verifies it is returned
+            currentEntry.setPriority(REINDEX_MAX_FAILURE_ATTEMPTS);
+            reindexQueueAPI.markAsFailed(currentEntry, "Testing failed record");
+
+            failedRecords = reindexQueueAPI.getFailedReindexRecords();
+            assertTrue(UtilMethods.isSet(failedRecords));
+            assertEquals(1, failedRecords.size());
+            assertEquals(recordId, failedRecords.get(0).getIdentToIndex());
+
+        }finally{
+            reindexQueueAPI.deleteReindexAndFailedRecords();
+        }
+    }
+
+
+    /**
+     * Method to test: {@link ReindexQueueAPI#getFailedReindexRecords()}
+     * Given Scenario: A reindex record with {@link com.dotmarketing.common.reindex.ReindexQueueFactory.Priority#REINDEX} , then a call to {@link ReindexQueueAPI#getFailedReindexRecords()} should not return this record
+     * ExpectedResult: None failed records returned
+     *
+     */
+    @Test
+    public void test_getFailedReindexRecords_shouldNotReturnAnyRecord() throws DotDataException {
+        final String recordId = UUIDGenerator.shorty();
+
+        //cleans up data first
+        reindexQueueAPI.deleteReindexAndFailedRecords();
+
+        try {
+            //adds a new entry and verifies it is not marked as failed
+            reindexQueueAPI.addReindexHighPriority(recordId);
+            List<ReindexEntry> failedRecords = reindexQueueAPI.getFailedReindexRecords();
+            assertFalse(UtilMethods.isSet(failedRecords));
+
+            //gets ReindexEntry from recordId
+            ReindexEntry currentEntry = getReindexEntry(recordId);
+
+            //forces the priority to be set to Priority.REINDEX
+            currentEntry.setPriority(Priority.REINDEX.ordinal() - 1);
+            reindexQueueAPI.markAsFailed(currentEntry, "Testing failed record");
+            failedRecords = reindexQueueAPI.getFailedReindexRecords();
+            assertFalse(UtilMethods.isSet(failedRecords));
+
+            currentEntry = getReindexEntry(recordId);
+            assertEquals(Priority.REINDEX.ordinal(), currentEntry.getPriority());
+
+        }finally{
+            reindexQueueAPI.deleteReindexAndFailedRecords();
+        }
+
+    }
+
+    /**
+     * Gets a {@link ReindexEntry} given its id
+     * @param recordId
+     * @return
+     * @throws DotDataException
+     */
+    private ReindexEntry getReindexEntry(String recordId) throws DotDataException {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'recordId' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5458294)